### ISSUE TITLE
[BUG] ComponentError.cancelled triggered on opening external application

### ISSUE THREAD

**Issue by cdelmer** [Adyen/adyen-ios#1591]
**Describe the bug**

We're seeing a spike of issues primarily affecting bcmc_mobile and vipps  where the components are sending an `ComponentError.cancelled` error, even though the payment gets authorized later through a notification.

From tracking events we were able to deduct that:
`AdyenActionComponent.handle(_ action: Action)` is called
`ActionComponentDelegate.didOpenExternalApplication(component: ActionComponent)` is not called
`ActionComponentDelegate.didFail(with error: Error, from component: ActionComponent)` is called with `ComponentError.cancelled` error.

**To Reproduce**
We can't figure out exactly in what configuration this happens, but from conversation with affected users:
1. Select bcmc_mobile 
2. **Authorise** payment in bank app
3. Our app opens again and user will see an error resulting from `ActionComponentDelegate.didFail(with error: Error, from component: ActionComponent)` method

**Screenshots**
We logged the error caught in `didFail` method in Crashlytics; the following screenshots are for one of the users affected by this issue
<img width="617" alt="Screenshot 2024-03-25 at 20 32 31" src="https://github.com/Adyen/adyen-ios/assets/104766117/2bfb513b-4c3b-4441-8099-6ddecc1d0a33">
![Screenshot 2024-03-25 at 09 42 41](https://github.com/Adyen/adyen-ios/assets/104766117/0a07ddd6-71e7-4e86-91fc-8187076d110c)

**Environment**
 - iOS Version: > 14
 - SDK Version: 5.6.0
 - Package management system: SPM
 - Xcode version: 15.2

**Comment by goergisn** [comment]
Hi @cdelmer,
From what I can see the `BrowserComponent` is invoked which should not be the case for the `bcmc_mobile` flow. 
Which might be caused by url of the RedirectAction being a http(s) url instead of a "deeplink".

To be able to dive deeper into this issue we might need some additional info.
Could you please get in touch with the [technical support team](https://www.adyen.help/hc/en-us/requests/new) to provide more information?

Best,
Alex

**Comment by aifrim-tgtg** [comment]
Hey @goergisn,

The technical support team has sent us to github. Please check case 03634911, contains a few psp references. We can provide more examples if needed.

Regards,

**Comment by goergisn** [comment]
Hi @aifrim-tgtg 
perfect I will check and get back to you asap :)

**Comment by aifrim-tgtg** [comment]
Hey @goergisn,

Any news on this?

Regards,

**Comment by goergisn** [comment]
Good morning @aifrim-tgtg,

We are looking into it but I don't have any updates on it yet.
I'll keep you posted here as soon as I know more.

In the meantime it might be interesting for the technical support team to know when this behavior started and/or if any update of api/sdk happened.

Best,
Alex

**Comment by aifrim-tgtg** [comment]
Yes, we migrated from 4.11.1 to 5.4.1 (and now to 5.7.0). The issue started with 5.4.1

**Comment by goergisn** [comment]
Hi @aifrim-tgtg,
We just released [`5.8.0`](https://github.com/Adyen/adyen-ios/releases/tag/5.8.0) which contains a fix that might have lead to the behavior that you reported.

**Comment by goergisn** [comment]
Closing the ticket for now - Please let us know if you can still observe the issue after upgrading :)

**Comment by cdelmer** [comment]
Hi,

We are still observing the issue.

Note that this is the same issue than https://github.com/Adyen/adyen-ios/issues/779 (which was also reported by someone from our organisation).

So we first started having the issue when updating from `4.3.0` to `4.4.0`
The issue was fixed in `4.7.0` and we stopped having it when we upgraded from `4.4.0` to `4.7.0`

It appeared again when we went from `4.11.1` to `5.4.1`

It remained since then, we are now on `5.15.0`

**Comment by erenbesel** [comment]
Hi @cdelmer thanks for the info! We have not made any potential changes for this on v4 versions, so that's odd. Could be iOS related.
 
For v5, while we investigate, if you can share with us more info such as reproducible steps and whether it occurs constantly or not, that would be great!

**Comment by cdelmer** [comment]
Hi @erenbesel,

Thanks for your answer.

As I mentioned, we first observed the issue in 2022, and it stopped when we started using `4.7.0` which contained the fix according to the other discussion I linked. We never had the issue again on `4.x`, it only started again in 2024 when we updated to `5.x`.

I can't provide reproducible steps, it seems to just happen sometimes, but the flow for users who encounter the error and users who don't look the same. 

The `ComponentError.cancelled` error is thrown for 24% of successful payments with Vipps and 40% of successful payments with BCMC mobile (note that they are successful because we started ignoring the error instead of triggering a cancellation in our system as we used to).

Hope this helps!
